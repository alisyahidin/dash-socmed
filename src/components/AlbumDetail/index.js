import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Modal } from 'react-bootstrap'

import { openModal } from '../../actions/album'

import PostLoader from '../Loader/PostLoader'

const AlbumDetail = ({
  album,
  openModal,
}) => (
  <Modal
    id="album-detail"
    size="lg"
    show={album.open}
    onHide={() => openModal(false)}
  >
    {album.error && <h3 className="w-100 text-center my-5">{album.error}</h3>}
    {album.loading && (
      <Modal.Body className="h-75 text-center">
        <PostLoader long />
      </Modal.Body>
    )}
    {album.data !== null && (
      <>
        <Modal.Header closeButton>
          <Modal.Title>
            <h4>{album.data.title}</h4>
          </Modal.Title>
        </Modal.Header>
        <Modal.Body className="px-4">
          {console.log(album)}
        </Modal.Body>
      </>
    )}
  </Modal>
)

const mapStateToProps = state => ({ ...state })

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    openModal
  }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(AlbumDetail)