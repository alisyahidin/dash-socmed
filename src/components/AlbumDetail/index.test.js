import React from 'react'
import { shallow, mount } from 'enzyme'
import AlbumDetail from './index'

it('render <AlbumDetail /> correctly', () => {
  const component = shallow(<AlbumDetail />)

  expect(component).toMatchSnapshot()
})