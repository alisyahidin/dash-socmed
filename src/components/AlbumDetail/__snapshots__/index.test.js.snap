// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <AlbumDetail /> correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
